src/routes/UserSettings/EditProfileTab/EditProfileTab.js

```diff
@@ -13,28 +13,99 @@ import { DEFAULT_BANNER } from 'store/models/Me'
 
 import './EditProfileTab.scss'
 
-const { object, func } = PropTypes
-
-const twitterPrompt = () => window.prompt('Please enter your twitter name.')
-export function linkedinPrompt (prompt) {
-  var url = window.prompt(prompt || 'Please enter the full url for your LinkedIn page.')
-  if (url) {
-    if (validateLinkedin(url)) {
-      return url
-    } else {
-      return linkedinPrompt('Invalid url. Please enter full url for your LinkedIn page.')
+const { object, func, string } = PropTypes
+
+export const socialLinkClicked = provider => {}
+
+const mapSocialProviderToKey = {
+  'twitter': 'twitterName',
+  'linkedin': 'linkedinUrl',
+  'facebook': 'facebookUrl'
+}
+
+/** LinkedIn Url */
+export const validateLinkedinUrl = url => url.match(/^(http(s)?:\/\/)?([\w]+\.)?linkedin\.com/)
+
+export const linkedinPrompt = () => {
+  let linkedinUrl = window.prompt('Please enter the full url for your LinkedIn page.')
+
+  if (linkedinUrl) {
+    while (!validateLinkedinUrl(linkedinUrl)) {
+      linkedinUrl = window.prompt('Invalid url. Please enter the full url for your LinkedIn page.')
     }
+
+    return linkedinUrl
   }
 }
+export  class SocialControl extends Component {
+  static propTypes = {
+    label: string,
+    provider: string,
+    value: string | null,
+    updateSocialSettings: func,
+    handleUnlink: func
+  }
 
-export const validateLinkedin = url => url.match(/^(http(s)?:\/\/)?([\w]+\.)?linkedin\.com/)
+  handleLinkClick () {
+    const { provider, updateSocialSetting } = this.props
 
+    switch (provider) {
+      case 'twitter': {
+        const twitterHandle = window.prompt('Please enter your twitter name.')
+        if (twitterHandle) {
+          updateSocialSetting({ key: mapSocialProviderToKey[provider], value: twitterHandle })
+        }
+        break;
+      }
+      case 'linkedin': {
+        const linkedinUrl = linkedinPrompt()
+        updateSocialSetting({ key: mapSocialProviderToKey[provider], value: linkedinUrl })
+      }
+      // case 'facebook': {
+      //   return onLink()
+      //   .then(({ error }) => {
+      //     if (error) return onChange(false)
+      //     return onChange(true)
+      //   })
+      // }
+    }
+  }
+
+  handleUnlinkClick () {
+    const { handleUnlinkAccount, updateSocialSetting, provider } = this.props
+
+    handleUnlinkAccount()
+    updateSocialSetting({ key: mapSocialProviderToKey[provider], value: '' })
```

Comment:
When users unlink their social accounts, do we want to set the value to `''` or `null`?

Reply:
null